shared-bindings/pulseio/PulseOut.c

```diff
@@ -41,10 +41,15 @@
 //|        pulsed signal consists of timed on and off periods. Unlike PWM, there is no set duration
 //|        for on and off pairs."""
 //|
-//|     def __init__(self, carrier: pwmio.PWMOut) -> None:
+//|     def __init__(self, pin: microcontroller.Pin, *, frequency: int = 38000, duty_cycle: int = 1 << 15) -> None:
 //|         """Create a PulseOut object associated with the given PWMout object.
 //|
-//|         :param ~pwmio.PWMOut carrier: PWMOut that is set to output on the desired pin.
+//|         :param ~microcontroller.Pin pin: PWMOut that is set to output on the desired pin.
+//|         :param int frequency: Carrier signal frequency in Hertz
+//|         :param int duty_cycle: 16-bit duty cycle of carrier frequency (0 - 65536)
+//|
+//|         For backwards compatibility, ``pin`` may be a PWMOut object used as the carrier. This
+//|         compatibility will be removed in CircuitPython 8.0.0.
 //|
```

Comment:
For now, before 8.0.0, should the type signature therefore be a `Union`? Otherwise people editing code using stubs from this may have complaints, I think.

Reply:
I did this deliberately to get folks using the new version. I think it's fair to not have stub support for deprecated types.